client/state/signup/steps/social-profiles/schema.ts

```diff
@@ -0,0 +1,23 @@
+export const SUPPORTED_SOCIAL_PROFILES = [
+	'FACEBOOK',
+	'TWITTER',
+	'INSTAGRAM',
+	'LINKEDIN',
+] as const;
+
+export type SocialProfile = typeof SUPPORTED_SOCIAL_PROFILES[ number ];
+
+export type SocialProfilesState = Record< SocialProfile, string >;
```

Comment:
This makes `SocialProfilesState` an object (Record) with possible keys restricted to `SUPPORTED_SOCIAL_PROFILES`
<img width="1025" alt="image" src="https://user-images.githubusercontent.com/5436027/157857108-a6c86071-b3cc-415d-bb44-09e90d116e5e.png">

Reply:
Sorry I was looking at the other PR and I am confused why do we need this abstraction? To restrict social profiles?

Reply:
IMO, this has a few benefits:
1. The `SocialProfile` type gives in-editor type hints and helps avoid typos when accessing this object by key (`FACEBOOK/TWITTER`)
2. It can be used to construct the initialState and schema.
3. Adding a social profile would be relatively simple since it would mean adding a key here and then adding display properties for that social profile in `client/signup/steps/social-profiles/social-profiles.tsx`

LMK if this answers your question!